### DISCUSSION TITLE
loading arbitrary waveforms as stimsets

### DISCUSSION THREAD

**Discussion by MaheshKarnani** [AllenInstitute/MIES#2307] [Q&A] [answered]
Hello,

We are trying to input user defined arbitrary waveforms as stimsets in MIES. Does anyone know how to do that?

Here's what we have tried so far: generated waveform in python, saved it in a .nwb file as 'stimulus' or 'icephys CurrentClampStimulusSeries', loaded stimsets from .nwb in MIES - but this did not work. Our best guess is that MIES stimsets are loaded from somewhere else within a .nwb file. 

Many thanks for your help!
-Mahesh

**Comment by timjarsky** [comment, marked as answer]
Hi @MaheshKarnani,

MIES supports loading custom stimsets (in a limited way) through the waveBuilder. 

1. import the array into Igor as a wave
2. Load the wave as an epoch from the `Load` tab of the wavebuilder.

![image](https://github.com/user-attachments/assets/00990363-9519-483a-bdc3-ff3d0b8a5a10)

This is old code, so you may need help with issues.  

Have you found that the waveBuilder cannot generate the desired stim set? We may be able to help with this as well.

Best, Tim

**Reply by MaheshKarnani** [reply]
Thanks this worked well! 

Just as a note to others: we needed to upsample stimuli to 200kHz to use this.

Best, Mahesh
